### PR TITLE
fix global & initial states

### PR DESCRIPTION
**Pull request by jacogr**
Really not sure why it has become an issue in the last couple of days, however due to the ordering of the subscriptions all hells starts breaking loose in the reducers as messages gets passed. (Components don't have all the info yet.) Originally reported by @tomusdrw  in https://github.com/ethcore/parity/issues/2133, the issue is actually a bit wider with balances as well.  (No doubt initialState should have been set right up-front as a minimum)

One thing on this - the whole Redux architecture makes me think more and more of micro-kernels vs Linux. The former is no doubt the best theoretical choice, the latter actually gets work done. Debugging these types of issues is an absolute crazy nightmare since you you have this black box of indirections instead of being close to your component code. Not fun to write or wire up, and definately not to debug. (I will continue following my approach of tying Redux in only when global state is absolutely needed, nowhere else.)

In short - 

- make sure sane defaults are always provided to initialState
- assume that the objects from the state is actually empty

Depends on Promise-based subscriptions (otherwise the 2 branches would have had conflicts with the better error reporting) - https://github.com/ethcore/parity/pull/2136 needs to be merged first

### CODE REVIEW

**Comment by derhuerst** [comment]
> Debugging these types of issues is an absolute crazy nightmare since you you have this black box of indirections instead of being close to your component code. Not fun to write or wire up, and definately not to debug.

Redux is still quite hard to wrap the head around imo, but throwing [redux-devtools](https://github.com/gaearon/redux-devtools) and [redux-logger](https://github.com/evgenyrodionov/redux-logger) in here.
